refactor: Deprecated user config from IConfig correctly

CarlSchwan opened the pull request:
Mark the methods in the interface deprecated instead of just the one in the implementation.

## Summary

Mark the methods in the interface deprecated instead of just the one in the implementation.

One of them was not on the public interface, so I directly removed it and ported the only usage.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)
- [x] [Labels added](https://github.com/nextcloud/server/labels) where applicable (ex: bug/enhancement, `3. to review`, feature component)
- [x] [Milestone added](https://github.com/nextcloud/server/milestones) for target branch/version (ex: 32.x for `stable32`)

CarlSchwan committed:
refactor: Deprecated user config from IConfig correctly

Mark the methods in the interface deprecated instead of just the one in
the implementation.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor(user_ldap): Port User\User to IUserConfig

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor(weather_status): Port WeatherStatusService to IUserConfig

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor(user_ldap): Port most of the remaining deprecated IConfig usage

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

provokateurin reviewed (changes requested):
LGTM, just one minor detail :sweat_smile:

CarlSchwan committed:
fix(IConfig): Change deprecation release to 33.0.0

The implementation was deprecated in 31.0.0 but since this wasn't done on
the interface, this is considered invalid. So change that to 33.0.0.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor: Make some code a bit more correct

- Use PHP_FLOAT_EPSILON for float comparaison
- Simplify some getValueBool code

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>
